static/app/components/events/featureFlags/setupIntegrationModal.tsx

```diff
@@ -0,0 +1,218 @@
+import {Fragment, useCallback, useState} from 'react';
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import {addSuccessMessage} from 'sentry/actionCreators/indicator';
+import type {ModalRenderProps} from 'sentry/actionCreators/modal';
+import Alert from 'sentry/components/alert';
+import {Button, LinkButton} from 'sentry/components/button';
+import ButtonBar from 'sentry/components/buttonBar';
+import SelectField from 'sentry/components/forms/fields/selectField';
+import type {Data} from 'sentry/components/forms/types';
+import TextCopyInput from 'sentry/components/textCopyInput';
+import {IconWarning} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import OrganizationStore from 'sentry/stores/organizationStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import {space} from 'sentry/styles/space';
+import {defined} from 'sentry/utils';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import useApi from 'sentry/utils/useApi';
+
+export type ChildrenProps<T> = {
+  Body: (props: {children: React.ReactNode}) => ReturnType<ModalRenderProps['Body']>;
+  Footer: () => ReturnType<ModalRenderProps['Footer']>;
+  Header: (props: {children: React.ReactNode}) => ReturnType<ModalRenderProps['Header']>;
+  state: T;
+};
+
+interface State {
+  provider: string;
+  url: string | undefined;
+}
+
+function useGenerateAuthToken({
+  state,
+  orgSlug,
+}: {
+  orgSlug: string | undefined;
+  state: State;
+}) {
+  const api = useApi();
+  const date = new Date().toISOString();
+
+  const createToken = async () =>
+    await api.requestPromise(`/organizations/${orgSlug}/org-auth-tokens/`, {
+      method: 'POST',
+      data: {
+        name: `${state.provider} Token ${date}`,
+      },
+    });
+
+  return {createToken};
+}
+
+export function SetupIntegrationModal<T extends Data>({
+  Header,
+  Body,
+  Footer,
+  closeModal,
+}: ModalRenderProps) {
+  const [state, setState] = useState<State>({
+    provider: 'LaunchDarkly',
+    url: undefined,
+  });
+  const {organization} = useLegacyStore(OrganizationStore);
+  const {createToken} = useGenerateAuthToken({state, orgSlug: organization?.slug});
+
+  const handleSubmit = useCallback(() => {
```

Comment:
```suggestion
  const handleClose = useCallback(() => {
```
When reading I thought the button that uses this submits something.

Nit: maybe we could put this inline with the button below?